core/src/main/java/org/apache/accumulo/core/client/rfile/RFileScanner.java

```diff
@@ -309,7 +303,7 @@ public Iterator<Entry<Key,Value>> iterator() {
 
       ClientIteratorEnvironment.Builder iterEnvBuilder = new ClientIteratorEnvironment.Builder()
           .withEnvironment(new RFileScannerEnvironmentImpl(opts)).withAuthorizations(opts.auths)
-          .withScope(IteratorScope.scan).withTableId(TABLE_ID);
+          .withScope(IteratorScope.scan);
```

Comment:
Would still need the `withTableId` here for the calls `env.getPluginEnv().getConfiguration(env.getTableId()`) (from javadocs for `IteratorEnvironment.getPluginEnv()`) so `env.getTableId()` doesn't throw an exception

Reply:
`RFileScannerEnvironmentImpl` overrides `ClientServiceEnvironmentImpl.getConfiguration(TableId)` and returns the `tableConf` regardless of the TableId.